samples/wifi/softap/src/main.c

```diff
@@ -325,19 +328,19 @@ static int configure_dhcp_server(void)
 		ret = func(__VA_ARGS__); \
 		if (ret) { \
 			LOG_ERR("Failed to configure %s", #func); \
-			return -1; \
+			return; \
```

Comment:
This depends on the caller, and I would recommend making the caller return the error always rather than void.